File: MultiCast_Delegate/MultiCast_Delegate/ViewController.m
//
//  ViewController.m
//  MultiCast_Delegate
//
//  Created by Rangel Cardoso Dias on 19/02/20.
//  Copyright © 2020 Rangel Cardoso Dias. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController{

    Robot *robot;
    Person *person;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.delegate = [[DelegatorHandler<MenssageDelegate> alloc] init];
    person = [[Person alloc] init];
    robot = [[Robot alloc] init];
    
    [self.delegate addDelegate:person];
    [self.delegate addDelegate:robot];
    
    
}
- (IBAction)buttonAction:(id)sender {
    [self.delegate dispachEvents];
}


@end

File: CoordinatorTest/CoordinatorTest/Controllers/ViewController.swift
//
//  ViewController.swift
//  CoordinatorTest
//
//  Created by Rangel Cardoso Dias on 21/02/20.
//  Copyright © 2020 Rangel Cardoso Dias. All rights reserved.
//

import UIKit


protocol FirstViewControllDelegate {
    
    func navigateToNextPage() -> Void
    
}
class ViewController: UIViewController {

    var delegate: FirstViewControllDelegate?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        
        
    }


    @IBAction func nativagateNextPage(_ sender: Any) {
        
        
        self.delegate?.navigateToNextPage()
        
        
    }
}

